orangewidget/settings.py

```diff
@@ -708,6 +708,17 @@ def __getstate__(self):
                 del state[nc]
         return state
 
+    def _stable_state(self):
+        state = self.__getstate__()
+        del state["time"]
+        return state
+
+    def __eq__(self, other):
+        return self._stable_state() == other._stable_state()
+
+    def __hash__(self):
+        return hash(list(self.__getstate__.items()))
```

Comment:
this does not work. __getstate__ is a function, you probably need to call it. list is not hashable, and neither are dictionaries (self.values). self.values can contain anything, not just hashable values.

Does context really need __hash__? Is it ever used in sets/keys of a dict?

Reply:
Please let us attribute this to late hour. The lesson is to always write a test.

I wrote `__hash__` because I thought I have to provide one, even if just to raise an exception (but then I thought I'd just implement a "real one"). I didn't know (or forgot) that defining `__eq__` already makes a class unhashable unless `__hash__` is defined explicitly.